.gitignore

```diff
@@ -1,3 +1,8 @@
+# Build subdirectories.
+/*build*
+!/build-aux
+!/build_msvc
+
```

Comment:
the mentioned `cmake-build-debug` and `cmake-build-debug-coverage` folders were created outside of build by default:
<img src="https://github.com/user-attachments/assets/3660fb5e-3e34-422b-a7d3-8a2aae3021e7">

Reply:
I'm not a CLion expert, but it seems inappropriate to create build artifacts anywhere outside of the CMake's designated binary tree. Is this configurable?

Reply:
yes, but it's not the default, should I open an issue for them?
<img src="https://github.com/user-attachments/assets/33ef2a4c-9a7a-4a1e-acd9-4993c6690c55">

Reply:
Opened https://youtrack.jetbrains.com/issue/CPP-39809/cmake-build-run-config-folders-should-be-created-inside-the-build-folder-by-default, thanks for checking

Reply:
I can see that "cmake-build-debug" is a configured name of the build directory for the "Debug" build type. It is OK to have multiple build directories simultaneously. The suggested pattern `/*build*` covers all such cases.

Reply:
> Opened https://youtrack.jetbrains.com/issue/CPP-39809/cmake-build-run-config-folders-should-be-created-inside-the-build-folder-by-default, thanks for checking

Just to clarify, based on https://github.com/hebasto/bitcoin/pull/302#discussion_r1703955762, I don't consider https://youtrack.jetbrains.com/issue/CPP-39809/cmake-build-run-config-folders-should-be-created-inside-the-build-folder-by-default to be a bug.

Reply:
Thanks, will let them decide, they have all the context here